libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2548,24 +2543,9 @@ void TypeChecker::requireLValue(Expression const& _expression)
 		}
 
 		if (auto memberAccess = dynamic_cast<MemberAccess const*>(&_expression))
-		{
 			if (auto structType = dynamic_cast<StructType const*>(type(memberAccess->expression())))
-			{
 				if (structType->dataStoredIn(DataLocation::CallData))
 					return "Calldata structs are read-only.";
-			}
-			else if (auto arrayType = dynamic_cast<ArrayType const*>(type(memberAccess->expression())))
-				if (memberAccess->memberName() == "length")
-					switch (arrayType->location())
-					{
-						case DataLocation::Memory:
-							return "Memory arrays cannot be resized.";
```

Comment:
Can we add a meaningful error for all cases?

Reply:
Yes, that makes sense and is also in line with @ekpyron's comment below :)